examples/firewall/firewall.c

```diff
@@ -0,0 +1,388 @@
+                return -1;
+        }
+        return optind;
+}
+
+/*
+ * This function displays stats. It uses ANSI terminal codes to clear
+ * screen when called. It is called from a single non-master
+ * thread in the server process, when the process is run with more
+ * than one lcore enabled.
+ */
+static void
+do_stats_display(void) {
+        const char clr[] = {27, '[', '2', 'J', '\0'};
+        const char topLeft[] = {27, '[', '1', ';', '1', 'H', '\0'};
+
+        /* Clear screen and move to top left */
+        printf("%s%s", clr, topLeft);
+        printf("Packets Dropped: %lu\n", stats.pkt_drop);
+        printf("Packets Accepted: %lu\n", stats.pkt_accept);
+        printf("Packets Total: %lu", stats.pkt_total);
+
+        printf("\n\n");
+}
+
+static int
+packet_handler(struct rte_mbuf *pkt, struct onvm_pkt_meta *meta, __attribute__((unused)) struct onvm_nf_info *nf_info) {
+        struct ipv4_hdr *ipv4_hdr;
+        static uint32_t counter = 0;
+        uint32_t rule = 0;
+        uint32_t track_ip = 0;
+
+        if (onvm_pkt_is_ipv4(pkt)) {
+                ipv4_hdr = onvm_pkt_ipv4_hdr(pkt);
+                int ret = rte_lpm_lookup(lpm_tbl, ipv4_hdr->src_addr, &rule);
+
+                if (ret) {
+                        meta->action = ONVM_NF_ACTION_DROP;
+                        stats.pkt_drop++;
+                        stats.pkt_total++;
+                        if (debug) {
+                                RTE_LOG(INFO, APP, "Packet from source IP %d has been dropped.\n", ipv4_hdr->src_addr);
+                        }
+                } else {
+                        switch (rule) {
+                                case 0:
+                                        meta->action = ONVM_NF_ACTION_TONF;
+                                        meta->destination = destination;
+                                        stats.pkt_accept++;
+                                        stats.pkt_total++;
+                                        if (debug) {
+                                                RTE_LOG(INFO, APP, "Packet from source IP %d has been accepted.\n",
+                                                        ipv4_hdr->src_addr);
+                                        }
+                                        break;
+                                default:
+                                        meta->action = ONVM_NF_ACTION_DROP;
+                                        stats.pkt_drop++;
+                                        stats.pkt_total++;
+                                        if (debug) {
+                                                RTE_LOG(INFO, APP, "Packet from source IP %d has been dropped.\n",
+                                                        ipv4_hdr->src_addr);
+                                        }
+                                        break;
+                        }
+                }
+        } else {
+                if (debug) {
```

Comment:
To not be extra verbose remove the { } braces from the one line `if (debug)` check